src/EnergyPlus/Plant/Solution/PlantLoopSolver.cc

```diff
@@ -168,215 +158,248 @@ namespace PlantLoopSolver {
-		PlantLoop( LoopNum ).LoopSide( ThisSide ).FlowLock = FlowUnlocked;
-		SimulateAllLoopSideBranches( LoopNum, ThisSide, ThisLoopSideFlow, FirstHVACIteration, LoopShutDownFlag );
-
-		// DSU? discussion/comments about loop solver/flow resolver interaction
-		// At this point, the components have been simulated.  They should have either:
-		//  - logged a massflowrequest
-		//  - or logged a MassFlowRate
-		// We need to decide what the components are going to do on FlowLock=0.
-		// If we want all control here at the solver level, the components just need to
-		//  log their MassFlowRate on their outlet nodes, or some other mechanism.
-		// Then the loop solver can scan the branch and get the max, and this will be the requested
-		//  flow rate for the branch.
-		// The loop solver will then set this as the branch outlet mass flow rate in preparation
-		//  for the flow resolver.
-		// The loop solver may need to do something to the inlet/outlet branch, but I'm not sure yet.
-		// The following comment block is what I had already thought of, and it may still make sense.
-
-		// Now that all the flow requests have been logged, we need to prepare them for the
-		//  flow resolver.  This will just take the requests and determine the desired flow
-		//  request for that branch according to pump placement, pump type, and other component
-		//  conditions.  In many cases, this will just be to simply take the max request from
-		//  the branch, which will already be within pumping limits for that flow path.
-		// We can then call the flow resolver to lock down branch inlet flow rates.
-		//DSU?
-
-		// The flow resolver takes information such as requested flows and min/max available flows and
-		//  sets the corrected flow on the inlet to each parallel branch
-		ResolveParallelFlows( LoopNum, ThisSide, ThisLoopSideFlow, FirstHVACIteration );
-		//  CALL PropagateResolvedFlow(LoopNum, ThisSide)
-
-		// Re-Initialize variables for this next pass
-		InitialDemandToLoopSetPointSAVED = InitialDemandToLoopSetPoint;
-		CurrentAlterationsToDemand = 0.0;
-		UpdatedDemandToLoopSetPoint = InitialDemandToLoopSetPoint;
+		// Do the actual simulation here every time
+		thisPlantLoop.loopSolver.DoFlowAndLoadSolutionPass(LoopNum, ThisSide, OtherSide, ThisSideInletNode, FirstHVACIteration);
 
-		// Now that flow rates have been resolved, we just need to set the flow lock status
-		//  flag, and resimulate.  During this simulation each component will still use the
-		//  SetFlowRequest routine, but this routine will also set the outlet flow rate
-		//  equal to the inlet flow rate, accoridng to flowlock logic.
-		PlantLoop( LoopNum ).LoopSide( ThisSide ).FlowLock = FlowLocked;
-		SimulateAllLoopSideBranches( LoopNum, ThisSide, ThisLoopSideFlow, FirstHVACIteration, LoopShutDownFlag );
+		// On constant speed branch pump loop sides we need to resimulate
+        if ( CurrentLoopSideIsConstantSpeedBranchPumped ) {
```

Comment:
However, if there are constant speed branch pumps, we now check to see if there were any branches that were on but shouldn't have been.  If so, we need to disable them and re-run the entire solution again to allow the flow to be dispatched properly without those extraneous branches available.